src/Mod/Path/PathScripts/PathProfileEdges.py

```diff
@@ -32,19 +32,25 @@
 
 from DraftGeomUtils import findWires
 from PySide import QtCore
+from Draft import makeWire
```

Comment:
I suggest you `import Draft`, and then use the function as `Draft.makeWire(...)` in the rest of your code.

Whoever reads your code will know immediately that these functions are not defined in this module, but are imported from other modules. Same with `DraftGeomUtils.findWires`.

Reply:
Sir,
     At the moment, I don't think we have had a proper introduction, here or on the FreeCAD forum.  I see you are quite involved in the FreeCAD project and have many, many contributions.  So, I say this in the most respectful way, I disagree with this suggestion.  
     As the FreeCAD project continues to grow, the demands on a user's computer resources grows.  At some point the devs must take reasonable steps to optimize code and performance.  Using the "from Module import Function" application seems to me a simple action with minimal hindrance to code development or understanding.
     As I understand Python, fewer resources(memory and parse time) are required if specific functions are imported as needed, rather than entire modules.  
     I believe anyone with much programming experience looking through the code, that wants to know if a function is defined within the current module, can press CTRL-F and search for the function by name, with a space or 'def ' proceeding it for Python.  Such a strategy will yield quick results.  Or, considering the vast amount of great help available in the forums, the same person can just ask if they cannot determine where  a function is defined.

Nonetheless, if the suggestion presented here is really a mandate, I'll conform, with a bit of heartburn likely to follow.

Respectfully,
Russell

Reply:
According to this page, the whole module is imported either way so it makes no difference to performance.  I'm not aware of a preferred style in the python community and haven't seen one emerge in the FreeCAD community to I'm ambivalent.  As long was we never EVER do 
"from module import *"   That will earn you a fast trip to my grumpy side.

https://softwareengineering.stackexchange.com/questions/187403/import-module-vs-from-module-import-function

Reply:
> As I understand Python, fewer resources(memory and parse time) are required if specific functions are imported as needed, rather than entire modules.

Unfortunately this is not true. When you import a function from a module, the entire module is read. You can verify this by adding a `print("something")` in the module, and then importing a completely different function from it. The Python parser goes through every single line; when it reaches the `print()` line, it will execute it, and you will see `"something"`.

['import module' vs. 'from module import function'](https://softwareengineering.stackexchange.com/questions/187403/import-module-vs-from-module-import-function)

>I believe anyone with much programming experience looking through the code, that wants to know if a function is defined within the current module, ...

It is a problem to make assumptions. As you very well know, many contributors to FreeCAD are not expert programmers. They are people who are learning along the way. That's exactly how Draft came to be. Yorik is also not a career programmer, he essentially learned programming while writing Draft and Arch, and he continues to learn to this day. This is the reason the code that we have today is a bit messy and not exactly well organized.

The [Zen of Python](https://www.python.org/dev/peps/pep-0020/) mentions that "explicit is better than implicit", and "namespaces are a good thing". What this means is that instead of assuming that another programmer will "figure it out", we should try to write the code in such a clear way that it's relatively easy for somebody else to understand it even if they are beginners. This is the strength of Python, that readability is important.

Another popular saying in programming is that "code is read more than it's written". Very often you have to go back and figure out why things work the way they do. If the code is sloppy, badly commented, convoluted, etc., then it's a pretty difficult task to do. Python has had its meteoric rise in the last 20 years precisely because it's relatively easy to read and thus learn.

Namespaces, the ability to organize code in sections is amazing. The way it's implemented in Python is great. This contrasts with the other side of FreeCAD, which is C++. There you basically have a global namespace. You don't know where a class or function comes from unless you have a good idea of the imported headers, those `<something.h>` files.

So, using `import Draft` is not a mandate, but it does improve general readability and robustsness of the code. I am right now going through Draft, fixing several things like that. So the suggestions I make are not meant to be absolute, but are reasonable in the context of improving the Python style in order to comply with [PEP8 guidelines](https://www.python.org/dev/peps/pep-0008/).

Reply:
I stand convinced and educated.  I will make the changes, and I will only import the entire module in future contributions.

Thanks gentlemen.

Reply:
I also want to point out that sliptonic quoted the same page I did!

And right now, in Draft and Arch, we absolutely use star imports `from module import *`. I'm trying to change this because we don't want to be in sliptonic's grumpy side!